doc/MANUAL.adoc

```diff
@@ -494,6 +494,15 @@ _<<Using ccache with other compiler wrappers>>_.
     symlinks). This is the default.
 *clang*::
     Clang-based compiler.
+[#config_compiler_type_clang_minimize_whitespace] 
+*clang_normalize_whitespace*::
+    Clang starting at version 14 supports the *-fminimize-whitespace*
+    option. The switch removes as much whitespace between tokens as
+    possible which reduces the chance of hashes of preprocessed
+    output being different only because of formatting changes. It is most
+    useful in <<the_unify_mode,unify mode>> where the *-P* switch
+    also removes newlines as much as possible, otherwise it only tries
+    to remove spaces between tokens on the same line.
```

Comment:
`compiler_type` seemed to be ccache's mechanism to identify a the set of flags supported by a compiler, `-fminimize-whitespace` would be one flavor of clang and therefore the most straightforward way to add another supported command line set. If gcc/others start supporting the flag, it could be made a bitflag, but at this point we cannot assume that gcc would also call there command line flag `-fminimize-whitespace`, but maybe `-fminify` in which case we'd indeed need a `gcc_minify`. What gcc will do is speculation at this point. There could be a more general mechanism to handle different versions of a compiler, but that's a lot more than needed for `-fminimize-whitespace` to work.

sccache compiles a small program to determine how the compiler identifies itself, including its version. The ccache way to do this seemed to be to follow the symlink. Clang's `bin/lang` and `bin/clang++` are symlinks to `bin/clang-${version}`, i.e. sufficient for our purpose. Unfortunately I only recently discovered that at least Ubuntu changed this in their distribution and does this the other way around: `clang-14` is a symlink to `/usr/lib/llvm-14/bin/clang`, so this patch at least needs to be extended to detect that pattern as well.

The best solution that also works with any compiler derived from clang would be a test invocation of `$cc -P -fminimize-whitespace` to see whether specifically this flag is support, but I assumed we'd not want to pay the cost of an additional program execution every time and I am not knowledgeable enough to add caching for the result.

When using unify mode there is no reason to no use `-fminimize-whitespace`, it is what the original unify mode did. Adding a configuration for enable it would make the ccache configuration more complex and most users would miss out on its advantages if had to be enabled explicitly.